ddcommon/src/azure_app_services.rs

```diff
@@ -155,6 +160,32 @@ impl AzureMetadata {
         })
     }
 
+    pub fn new<T: QueryEnv>(query: T) -> Option<Self> {
+        let is_relevant = query
+            .get_var(SERVICE_CONTEXT)
+            .map(|s| s.to_bool())
+            .unwrap_or(false);
+
+        if !is_relevant {
+            return None;
+        }
+
+        AzureMetadata::build_metadata(query)
+    }
+
+    pub fn new_function<T: QueryEnv>(query: T) -> Option<Self> {
+        let is_relevant = matches!(
```

Comment:
I suspect this could be re-written as a match statement, although the `matches!` macro makes that a little weird.

Something like:
```rust
match matches!(...) {
  true => AzureMetadata::build_metadata(query),
  false => None
}
```

Reply:
Done!